### PR TITLE
[Phase-2] Fix usage of L1 egamma seeds in Phase-2 HLT

### PR DESCRIPTION
**Pull request by SohamBhattacharya**
This PR fixes the usage of L1 egamma seeds in Phase-2 HLT.
See details of the fix in these [slides](https://indico.cern.ch/event/1322372/contributions/5564948/attachments/2708940/4703652/2023-09-05_HLT-upgrade-meeting.pdf#page=10).

Tagging Phase-2 Egamma HLT contacts/validators: @ravindkv @skeshri

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by SohamBhattacharya** [comment]
> +1
> 
> **Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-9a6972/34765/summary.html **COMMIT**: [a96c90c](https://github.com/cms-sw/cmssw/commit/a96c90c81f11a669e86a6da78e3df0f69a684bbb) **CMSSW**: CMSSW_13_3_X_2023-09-14-1100/el8_amd64_gcc11 **User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/42791/34765/install.sh` to create a dev area with all the needed externals and cmssw changes.
> ## Comparison Summary
> 
> **Summary**:
> 
>     * You potentially removed 5 lines from the logs
> 
>     * Reco comparison results: 12 differences found in the comparisons
> 
>     * DQMHistoTests: Total files compared: 50
> 
>     * DQMHistoTests: Total histograms compared: 3348648
> 
>     * DQMHistoTests: Total failures: 3
> 
>     * DQMHistoTests: Total nulls: 0
> 
>     * DQMHistoTests: Total successes: 3348623
> 
>     * DQMHistoTests: Total skipped: 22
> 
>     * DQMHistoTests: Total Missing objects: 0
> 
>     * DQMHistoSizes: Histogram memory added: 0.0 KiB( 49 files compared)
> 
>     * Checked 214 log files, 167 edm output root files, 50 DQM output files
> 
>     * [TriggerResults](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults): found differences in 4 / 48 workflows

Changes to TriggerResults in [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults/24834.0_TTbar_14TeV+2026D98) and [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults/24834.911_TTbar_14TeV+2026D98_DD4hep) look expected.

File: HLTrigger/HLTfilters/plugins/L1TTkEleFilter.cc [hunk 1]
```diff
@@ -147,8 +151,8 @@ bool L1TTkEleFilter::hltFilter(edm::Event& iEvent,
         passIsolation = true;
     }
 
-    if (offlinePt >= min_Pt_ && itkEle->eta() <= max_Eta_ && itkEle->eta() >= min_Eta_ && passQuality &&
-        passIsolation) {
+    if (offlinePt >= min_Pt_ && std::fabs(itkEle->eta()) < max_AbsEta1_ && std::fabs(itkEle->eta()) >= min_AbsEta1_ &&
```
Comment (mmusich) [inline HLTrigger/HLTfilters/plugins/L1TTkEleFilter.cc:154]: elsewhere in this file `std::abs` is used

https://github.com/cms-sw/cmssw/blob/a96c90c81f11a669e86a6da78e3df0f69a684bbb/HLTrigger/HLTfilters/plugins/L1TTkEleFilter.cc#L148-L149

I think that's the preferred option.

File: HLTrigger/HLTfilters/plugins/L1TTkEmFilter.cc [hunk 2]
```diff
@@ -147,7 +152,8 @@ bool L1TTkEmFilter::hltFilter(edm::Event& iEvent,
         passIsolation = true;
     }
 
-    if (offlinePt >= min_Pt_ && itkEm->eta() <= max_Eta_ && itkEm->eta() >= min_Eta_ && passQuality && passIsolation) {
+    if (offlinePt >= min_Pt_ && std::fabs(itkEm->eta()) < max_AbsEta1_ && std::fabs(itkEm->eta()) >= min_AbsEta1_ &&
```
Comment (mmusich) [inline HLTrigger/HLTfilters/plugins/L1TTkEmFilter.cc:155]: elsewhere in this file `std::abs` is used

https://github.com/cms-sw/cmssw/blob/a96c90c81f11a669e86a6da78e3df0f69a684bbb/HLTrigger/HLTfilters/plugins/L1TTkEmFilter.cc#L181-L182

I think that's the preferred option.

File: HLTrigger/HLTfilters/plugins/L1TTkEmFilter.cc [hunk 3]
```diff
@@ -177,7 +184,8 @@ bool L1TTkEmFilter::hltFilter(edm::Event& iEvent,
         passIsolation = true;
     }
 
-    if (offlinePt >= min_Pt_ && itkEm->eta() <= max_Eta_ && itkEm->eta() >= min_Eta_ && passQuality && passIsolation) {
+    if (offlinePt >= min_Pt_ && std::fabs(itkEm->eta()) <= max_AbsEta2_ && std::fabs(itkEm->eta()) >= min_AbsEta2_ &&
```
Comment (mmusich) [inline HLTrigger/HLTfilters/plugins/L1TTkEmFilter.cc:187]: see above .

File: RecoHGCal/TICL/plugins/SeedingRegionByL1.cc [hunk 4]
```diff
@@ -35,9 +36,10 @@ void ticl::SeedingRegionByL1::makeRegions(const edm::Event &ev,
   for (size_t indx = 0; indx < (*l1TrkEms).size(); indx++) {
     const auto &l1TrkEm = (*l1TrkEms)[indx];
     double offlinePt = this->tkEmOfflineEt(l1TrkEm.pt());
-    if ((offlinePt < minPt_) || (std::abs(l1TrkEm.eta()) < minAbsEta_) || (std::abs(l1TrkEm.eta()) > maxAbsEta_) ||
-        (l1TrkEm.EGRef()->hwQual() != quality_)) {
-      continue;
+    if ((offlinePt < minPt_) || (std::abs(l1TrkEm.eta()) < minAbsEta_) || (std::abs(l1TrkEm.eta()) > maxAbsEta_)) {
```
Comment (mmusich) [inline RecoHGCal/TICL/plugins/SeedingRegionByL1.cc:39]: this piece of code is a bit hard to read, I would suggest splitting conditions in dedicated booleans and then compose the final `if` accordingly (no strong push though).

**Review by mmusich — COMMENTED** [review]
> Changes to TriggerResults in [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults/24834.0_TTbar_14TeV+2026D98) and [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults/24834.911_TTbar_14TeV+2026D98_DD4hep) look expected.

I would be nice if also for the e/g objects a dedicated DQM / validation monitoring could be added (see standing issue https://github.com/cms-sw/cmssw/issues/39362)

**Comment by SohamBhattacharya** [comment]
> > Changes to TriggerResults in [24834.0](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults/24834.0_TTbar_14TeV+2026D98) and [24834.911](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-09-14-1100+9a6972/58880/triggerResults/24834.911_TTbar_14TeV+2026D98_DD4hep) look expected.
> 
> I would be nice if also for the e/g objects a dedicated DQM / validation monitoring could be added (see standing issue #39362)

Indeed, that's at the top of our to-do list of post Annual Review tasks.

**Comment by mmusich** [comment]
@cmsbuild, please test

**Comment by mmusich** [comment]
type bug-fix

**Comment by swagata87** [comment]
Thank you Soham for the fix

**Comment by mmusich** [comment]
+hlt
   * bug-fix PR is inline with initial description and follow-up review
   * a backport of this PR (https://github.com/cms-sw/cmssw/pull/42792) is going to be needed in CMSSW_13_1_X in order to fulfill the goals of the Upgrade HLT group for the 2023 AR
   * as discussed in https://github.com/cms-sw/cmssw/issues/39362#issuecomment-1721023778 and following addition of e/g objects monitoring will follow-up

**Comment by jfernan2** [comment]
+1

**Comment by srimanob** [comment]
+upgrade

**Comment by mmusich** [comment]
@cms-sw/orp-l2 please take a look to this.

**Comment by antoniovilela** [comment]
+1

**Pull request merged**